tests/dialects/test_postgres.py

```diff
@@ -99,6 +99,9 @@ def test_postgres(self):
         self.validate_identity(
             "SELECT * FROM JSON_ARRAY_ELEMENTS('[1,true, [2,false]]') WITH ORDINALITY AS kv_json(a, b)"
         )
+        self.validate_identity(
+            """SELECT * FROM table, ROWS FROM (FUNC1(col1) AS alias1("col1" TEXT), FUNC2(col2) AS alias2("col2" INT)) WITH ORDINALITY AS alias3("col3" INT, "col4" TEXT)"""
+        )
```

Comment:
This needs better test coverage, e.g. try `SELECT * FROM ROWS FROM (...)`. We can also have one without the ordinality.